crates/optimism/node/src/txpool.rs

```diff
@@ -445,52 +531,54 @@ pub struct OpL1BlockInfo {
     timestamp: AtomicU64,
 }
 
-#[cfg(test)]
-mod tests {
-    use crate::txpool::{OpPooledTransaction, OpTransactionValidator};
-    use alloy_eips::eip2718::Encodable2718;
-    use alloy_primitives::{PrimitiveSignature as Signature, TxKind, U256};
-    use op_alloy_consensus::{OpTypedTransaction, TxDeposit};
-    use reth_chainspec::MAINNET;
-    use reth_optimism_primitives::OpTransactionSigned;
-    use reth_primitives::Recovered;
-    use reth_provider::test_utils::MockEthProvider;
-    use reth_transaction_pool::{
-        blobstore::InMemoryBlobStore, validate::EthTransactionValidatorBuilder, TransactionOrigin,
-        TransactionValidationOutcome,
-    };
-    #[test]
-    fn validate_optimism_transaction() {
-        let client = MockEthProvider::default();
-        let validator = EthTransactionValidatorBuilder::new(MAINNET.clone())
-            .no_shanghai()
-            .no_cancun()
-            .build(client, InMemoryBlobStore::default());
-        let validator = OpTransactionValidator::new(validator);
-
-        let origin = TransactionOrigin::External;
-        let signer = Default::default();
-        let deposit_tx = OpTypedTransaction::Deposit(TxDeposit {
-            source_hash: Default::default(),
-            from: signer,
-            to: TxKind::Create,
-            mint: None,
-            value: U256::ZERO,
-            gas_limit: 0,
-            is_system_transaction: false,
-            input: Default::default(),
-        });
-        let signature = Signature::test_signature();
-        let signed_tx = OpTransactionSigned::new_unhashed(deposit_tx, signature);
-        let signed_recovered = Recovered::new_unchecked(signed_tx, signer);
-        let len = signed_recovered.encode_2718_len();
-        let pooled_tx = OpPooledTransaction::new(signed_recovered, len);
-        let outcome = validator.validate_one(origin, pooled_tx);
-
-        let err = match outcome {
-            TransactionValidationOutcome::Invalid(_, err) => err,
-            _ => panic!("Expected invalid transaction"),
-        };
-        assert_eq!(err.to_string(), "transaction type not supported");
-    }
-}
+// TODO(refcell): fix this in a follow on PR
```

Comment:
Will address this in a follow-on PR since this will require making a wrapper around the `MockEthProvider<OpTransactionSigned>` and implement traits on that wrapper with the specified tx.

Reply:
cc @emhane

Reply:
Wrote up a ticket to track this: https://github.com/ethereum-optimism/op-reth/issues/45

Reply:
Fixed in https://github.com/ethereum-optimism/op-reth/pull/47